### PR TITLE
Fix docker image check for faster OCR deployments

### PR DESCRIPTION
**Pull request by derekadombek**
## Description
We temporarily disabled the docker image checker cause it was bugging out, just to get the deployment automation moving forward. Now we have the time to fix it.

## Related Issues
[Link any related issues or tasks from your project management system.]

## Checklist
- [x] The title of this PR is descriptive and concise.
- [x] My changes follow the style guidelines of this project.
- [x] I have added or updated test cases to cover my changes.
- [ ] I've let the team know about this PR by linking it in the review channel

### CODE REVIEW

File: .github/workflows/build-deploy-frontend.yml [hunk 1]
```diff
@@ -24,55 +19,34 @@ permissions:
...
-          tar -C ./dist/ -czf ./client.tgz .
-          echo "::endgroup::"
-      - name: Upload production-ready build files
-        uses: actions/upload-artifact@v4
-        with:
-          name: production-files
-          path: ./frontend/client.tgz
+          api-endpoint: https://reportvision-ocr-${{ inputs.deploy-env }}.azurewebsites.net/
+          frontend-tarball: ./frontend.tgz
+          frontend-path: ./frontend
```
Comment (derekadombek) [inline .github/workflows/build-deploy-frontend.yml:31]: def doing a a lot of scope creeping on this PR but its a needed clean-up.

File: .github/workflows/build-deploy-frontend.yml [hunk 2]
```diff
@@ -24,55 +19,34 @@ permissions:
...
-          echo "::group::Create application archive"
-          tar -C ./dist/ -czf ./client.tgz .
-          echo "::endgroup::"
-      - name: Upload production-ready build files
-        uses: actions/upload-artifact@v4
-        with:
-          name: production-files
-          path: ./frontend/client.tgz
+          api-endpoint: https://reportvision-ocr-${{ inputs.deploy-env }}.azurewebsites.net/
+          frontend-tarball: ./frontend.tgz
```
Comment (marycrawford) [inline .github/workflows/build-deploy-frontend.yml:30]: Is the npm packages being installed in the `./frontend.tgz` file?

**Review by marycrawford — COMMENTED** [review]
Great move to clean up the environments, to be consistent with the terraform code.

File: .github/workflows/build-deploy-frontend.yml (hunk 2 above)
Comment (derekadombek) [reply .github/workflows/build-deploy-frontend.yml:30]: thats correct. at least with GHA, packaging all the frontend files into a tarball/zip makes downloading from other jobs smoother.

**Review by marycrawford — APPROVED** [review]
LGTM
